KODAK-Portrait-3D-Printer-Support-added

Jojo09092007 opened the pull request:
# Description
I did not want to use the old Kodak Slicer (3.6.6) anymore, so I integrated it into the new Cura.
I added the KODAK Portrait 3D Printer itself, the platform (STL-File, i reduced its file size), extruder1, extruder2 and its generic materials (which are now removed from the pull request). If you want them too contact me!
All the Files are from https://github.com/Smartinternational/KodakCura and are 100% working.
## Type of change

- [x] Printer definition file(s)

# How Has This Been Tested?

I tested it on my own KODAK Portrait 3D Printer, it works great.

- I tested it with a one color print, which worked great.
- I tested it with a two color print, which worked fine, if you merge the Models in Cura.

**Test Configuration**:
* Operating System: Windows 11 Pro for Workstations (Version 10.0.22631)

# Checklist:

- [x] My code follows the style guidelines of this project as described in [UltiMaker Meta](https://github.com/Ultimaker/Meta) and [Cura QML best practices](https://github.com/Ultimaker/Cura/wiki/QML-Best-Practices)
- [x] I have read the [Contribution guide](https://github.com/Ultimaker/Cura/blob/main/CONTRIBUTING.md) 
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have uploaded any files required to test this change

Jojo09092007 committed:
Create Kodak-Portrait.def.json

Addes

Jojo09092007 committed:
Rename Kodak-Portrait.def.json to kodak_portrait.def.json

Renamed

Jojo09092007 committed:
Create kodak_portrait_extruder_0

Jojo09092007 committed:
Rename kodak_portrait_extruder_0 to kodak_portrait_extruder_0.def.json

Jojo09092007 committed:
Create kodak_portrait_extruder_1.def.json

Jojo09092007 committed:
Update kodak_portrait_extruder_0.def.json

Added content

Jojo09092007 committed:
Update kodak_portrait_extruder_1.def.json

Added content

Jojo09092007 committed:
Rename kodak_portrait.def.json to kodakportrait.def.json

Jojo09092007 committed:
Create Default.cfg

Jojo09092007 committed:
Rename Default.cfg to Default.inst.cfg

Jojo09092007 committed:
Rename Default.inst.cfg to normal.inst.cfg

Jojo09092007 committed:
Add files via upload

It's the file for the platform

Jojo09092007 committed:
Aktualisieren von normal.inst.cfg

Jojo09092007 committed:
Rename normal.inst.cfg to kodak_global_normal.inst.cfg

shancui0604 commented:
Hello,I also want to add my definition into curat,I try several times but  failed.I met the same failed with you,l.Do you know the reason about it?

Jojo09092007 commented:
There was a change in the main cura branch, which completly destroyed the bot.

Jojo09092007 committed:
Update kodak_portrait_extruder_0.def.json

Jojo09092007 committed:
Update kodak_portrait_extruder_1.def.json

Jojo09092007 committed:
Update kodakportrait.def.json

jellespijker committed:
Merge branch 'main' into main

Jojo09092007 committed:
Merge branch 'main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Delete resources/meshes/kodak_portrait_platform.stl

Jojo09092007 committed:
Add files via upload

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Update kodak_global_normal.inst.cfg

Jojo09092007 committed:
Applied printer-linter format

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

KlasKronander commented:
Hi @Jojo09092007. Can you provide a bit more information about how this works? Is it a feature-complete replacement of the legacy Kodak Slicer 3.6.6. (which seems unavailable these days)?

Is there any particular reason this isn't merged yet? How does this relate to the two closed PRs #18918 and #19040 ?

Let me know if I can help.

Jojo09092007 commented:
@KlasKronander Yes it is a 1:1 copy of the Kodak Portrait config files + buildplate stl-files. It is now fully tested, should probably update description, Thanks for the attention.

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 commented:
> Hi @Jojo09092007. Can you provide a bit more information about how this works? Is it a feature-complete replacement of the legacy Kodak Slicer 3.6.6. (which seems unavailable these days)?
> 
> Is there any particular reason this isn't merged yet? How does this relate to the two closed PRs #18918 and #19040 ?
> 
> Let me know if I can help.

The Slicer is still available i dont think i can share my personal google drive link, but it exists. If you need it i might make are repo for it. Also read the description again, there you can see the original repo, where I got everything from.

macbosse commented:
I have the same Kodak 3D Portrait printer and am looking for a solution to keep it running now that the manufacturer has gone out of business or at least stopped supporting it.
What is the status? Is the printer now fully Cura compatible by default or does something still need to be installed?

Jojo09092007 commented:
> I have the same Kodak 3D Portrait printer and am looking for a solution to keep it running now that the manufacturer has gone out of business or at least stopped supporting it.
> What is the status? Is the printer now fully Cura compatible by default or does something still need to be installed? 

It works perfectly, my process was the following:
- I found a spanish version of the Kodak Slicer
- Copied the Printer .josn and the 2 extruders .json
- Lastly I copied the buildplate .stl, and scaled it down under 1mb or whatever(I don't remember), I followed a tutorial to align it graphically in Cura
- And then I testprinted and it worked perfectly, though you might need to ajust the buildplate limits, you can't print stuff on the complete edge, otherwise it will become a vibrator or switch the extruder.
- PROFIT!!!
You can do it yourself, I included a link to the repo of the original slicer, on my repo. I now am also trying to get Mainsail OS running on my Kodak,which is sadly still not done.

Jojo09092007 commented:
> I have the same Kodak 3D Portrait printer and am looking for a solution to keep it running now that the manufacturer has gone out of business or at least stopped supporting it. What is the status? Is the printer now fully Cura compatible by default or does something still need to be installed?

I created a portable version for you just extract it: https://drive.google.com/drive/folders/1rogGV3M3iKcNA_XlPY7UgMDBn3KxC7DO?usp=drive_link

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 commented:
> I have the same Kodak 3D Portrait printer and am looking for a solution to keep it running now that the manufacturer has gone out of business or at least stopped supporting it.
> What is the status? Is the printer now fully Cura compatible by default or does something still need to be installed? 

Did it work for you?

KlasKronander commented:
I just tried it today for the first time. It looks weird though, as if it will try to print lower than the build plate. See attached screenshots. 

![image](https://github.com/user-attachments/assets/7543f0ba-2a0f-4009-bbcd-022df1465a1a)
![image](https://github.com/user-attachments/assets/20291137-df27-454a-954c-6b1fc97c47be)

Jojo09092007 commented:
> I just tried it today for the first time. It looks weird though, as if it will try to print lower than the build plate. See attached screenshots. 
> 
> ![image](https://github.com/user-attachments/assets/7543f0ba-2a0f-4009-bbcd-022df1465a1a)
> ![image](https://github.com/user-attachments/assets/20291137-df27-454a-954c-6b1fc97c47be)
> 

That is a visual bug, you can adjust that in the printer config file, will fix it later.

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Applied printer-linter format

KlasKronander commented:
Thanks for all your work, really happy to be able to use Cura with my kodak printer. As you said, it was just a visual bug and the parts printed fine!

Are you sure that these last commits address the visual bug though? Looking at those commits, I see nothing that changes any kodak configuration files? Looks like just a merge from upstream plus a linter commit. I assume the required change is some configuration file about the build plate? I am a bit weary about recompiling with this new version due to the transition to conan2. 

If I wanted to get rid of this visual bug just locally on my branch, which file is it that needs updating?

Jojo09092007 commented:
> Thanks for all your work, really happy to be able to use Cura with my kodak printer! As you said, it was just a visual bug and the parts printed fine!
> 
> Are you sure that these last commits address the visual bug though? Looking at those commits, I see nothing that changes any kodak configuration files? Looks like just a merge from upstream plus a linter commit. I assume the required change is some configuration file about the build plate? I am a bit weary about recompiling with this new version due to the transition to conan2. 
> 
> If I wanted to get rid of this visual bug just locally on my branch, which file is it that needs updating?
> 
> 

It is located in resources/definition/kodakportait.json ![Screenshot_2025-02-03-07-07-25-48_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/user-attachments/assets/bb154637-891f-4af8-ad6b-6732a2126922)

These are the updated values.

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 commented:
> Thanks for all your work, really happy to be able to use Cura with my kodak printer! As you said, it was just a visual bug and the parts printed fine!
> 
> Are you sure that these last commits address the visual bug though? Looking at those commits, I see nothing that changes any kodak configuration files? Looks like just a merge from upstream plus a linter commit. I assume the required change is some configuration file about the build plate? I am a bit weary about recompiling with this new version due to the transition to conan2. 
> 
> If I wanted to get rid of this visual bug just locally on my branch, which file is it that needs updating?
> 
> 

I fixed everything in my github version, the downloadable Cura is just a messy port from the original files, I changed that in the new downloadable version, I hope the printer will be added eventually, I am now waiting almost 2 years for this, my patience are running thin, but I can do nothing.

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

lucahttp commented:
Hi Guys, I can help in something? I have just buyed this printer and compared with other printers (like bambulab) lacks a lot of software

I was looking to move away from this strange software and go for something like octoprint or astroprint

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'main' into main

Jojo09092007 committed:
Removed depricated "category" from kodakportrait.def.json

Jojo09092007 committed:
Update kodak_portrait_extruder_0.def.json to be same same but different

Jojo09092007 committed:
Apply printer-linter format

Jojo09092007 committed:
Update kodak_portrait_extruder_1.def.json to be same same but different

Jojo09092007 committed:
Apply printer-linter format

Jojo09092007 commented:
Now "fixed" all the stuff that needs updating, because some stuff (that works now) didn't work back then.

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

Jojo09092007 committed:
Merge branch 'Ultimaker:main' into main

theisishappening commented:
So, once I've downloaded this files from the google drive link, what do I do with them, how do I add this to Cura? Sorry simple question I know but can't figure this out.

Jojo09092007 commented:
> So, once I've downloaded this files from the google drive link, what do I do with them, how do I add this to Cura? Sorry simple question I know but can't figure this out. 
> 

Well my google drive link is a standalone version of Cura 4, you just need to run the cura. exe in the folder and then in the printer add tap, the Kodak Portrait should appear.

3djammix05 commented:
I'm having a issue implementing this is there a step by step totorial. I attempted to add in the files by name to the cura folders and this just crashed cura. I see all the successful people comment what I'm I missing

Jojo09092007 commented:
> I'm having a issue implementing this is there a step by step totorial. I attempted to add in the files by name to the cura folders and this just crashed cura. I see all the successful people comment what I'm I missing 

You need to only copy the kodak files into your cura appdata folder:
Dir: resources>definitions and copy the kodakportrait file in the according folder.(Likely in %appdata% in Windows)(I don't use the really modern versions of cura) 
And Dir: resources>extruders (copy the 2 kodak def to according location) 
And Dir: resources>meshes (search the kodak stl and copy it)
